src/_P103_Atlas_EZO_pH_ORP_EC.ino

```diff
@@ -0,0 +1,641 @@
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_single\").onclick=function() {document.getElementById(\"Plugin_103_enable_cal_dry\").checked = false;document.getElementById(\"Plugin_103_enable_cal_L\").checked = false;;document.getElementById(\"Plugin_103_enable_cal_H\").checked = false;};</script>"));
+            addFormNumericBox(F("Ref EC"), F("Plugin_103_ref_cal_single"), PCONFIG(3));
+            addUnit(F("&micro;S"));
+
+            addRowLabel(F("<strong>Low calibration</strong>"));
+            addFormCheckBox(F("Enable"),
+                            F(
+                                "Plugin_103_enable_cal_L' onClick='document.getElementById(\"Plugin_103_enable_cal_dry\").checked = false;document.getElementById(\"Plugin_103_enable_cal_single\").checked = false;;document.getElementById(\"Plugin_103_enable_cal_H\").checked = false;"),
+                            false);
+            addHtml(F(
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_L\").onclick=function() {document.getElementById(\"Plugin_103_enable_cal_dry\").checked = false;document.getElementById(\"Plugin_103_enable_cal_single\").checked = false;;document.getElementById(\"Plugin_103_enable_cal_H\").checked = false;};</script>"));
+            addFormNumericBox(F("Ref EC"), F("Plugin_103_ref_cal_L"), PCONFIG(4));
+            addUnit(F("&micro;S"));
+
+            addRowLabel(F("<strong>High calibration</strong>"));
+            addFormCheckBox(F("Enable"), F("Plugin_103_enable_cal_H"), false);
+            addHtml(F(
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_H\").onclick=function() {document.getElementById(\"Plugin_103_enable_cal_dry\").checked = false;document.getElementById(\"Plugin_103_enable_cal_single\").checked = false;;document.getElementById(\"Plugin_103_enable_cal_L\").checked = false;};</script>"));
+            addFormNumericBox(F("Ref EC"), F("Plugin_103_ref_cal_H"), PCONFIG(5));
+            addUnit(F("&micro;S"));
+
+            status = _P103_send_I2C_command(I2Cchoice, F("Cal,?"), sensordata);
+
+            if (status) {
+                switch (sensordata[5]) {
+                case '0':
+                    addFormNote(F("<span style='color:red'>Calibration needed</span>"));
+                    break;
+                case '1':
+                    addFormNote(F("<span style='color:green'>Single point calibration ok</span>"));
+                    break;
+                case '2':
+                    addFormNote(F("<span style='color:green'>Two points calibration ok</span>"));
+                    break;
+                }
+            }
+        }
+        break;
+      }
+
+        // Temperature compensation
+      if(board_type == PH || board_type == EC) {
+        
+        addFormSubHeader(F("Temperature compensation"));
+        char deviceTemperatureTemplate[40];
```

Comment:
Make sure it is zero-terminated before constructing a `String` object from it.
For example call the macro `ZERO_TERMINATE(deviceTemperatureTemplate);` after filling the array.

Reply:
OK